pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -663,13 +663,42 @@ func (l *LibvirtDomainManager) getGuestTimeContext() context.Context {
 
 // FinalizeVirtualMachineMigration apply all post-migration changes on the domain.
 func (l *LibvirtDomainManager) FinalizeVirtualMachineMigration(vmi *v1.VirtualMachineInstance) error {
+	domainName := api.VMINamespaceKeyFunc(vmi)
+	virDomain, err := l.virConn.LookupDomainByName(domainName)
```

Comment:
Reading the domain is for the hotplug only, so I believe it should be part of `hotPlugHostDevices` and after locking `domainModifyLock` (isn't that the whole purpose on this lock?).

Reply:
I moved "reading domain" part to `hotPlugHostDevices`